tools/build_defs/repo/http.bzl

```diff
@@ -139,6 +140,8 @@ def _http_archive_impl(ctx):
     auth = get_auth(ctx, all_urls)
 
     download_info = ctx.download_and_extract(
+        # TODO(fzakaria): all_urls here has the remote_patch URL which is incorrect
```

Comment:
@fmeum I think this is a bug -- should I cut one?

Reply:
Yeah, this seems wrong to me. This was likely an oversight in https://github.com/bazelbuild/bazel/commit/445dfab8fdd7671fe6206ea938ef119c228019f0, which just wanted to get this into `get_auth`.

CC @Wyverald